annif/project.py

```diff
@@ -295,7 +296,12 @@ def learn(
         beparams = backend_params.get(self.backend.backend_id, {})
         corpus = self.transform.transform_corpus(corpus)
         if isinstance(self.backend, annif.backend.backend.AnnifLearningBackend):
-            self.backend.learn(corpus, beparams)
+            if annif.util.boolean(self.config.get("allow_learn", False)):
+                self.backend.learn(corpus, beparams)
+            else:
+                raise OperationFailedException(
```

Comment:
How about defining and raising here a new type of exception such as OperationNotEnabledException (or something along those lines)? This would indicate that the operation failed not because of a technical failure, but due to a configuration setting.